### PR TITLE
Set learners name on search to first and last name pair

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
- fixes https://github.com/mitodl/micromasters/issues/1712

#### What's this PR do?
- On search page, instead of showing preferred name of learners, it shows first and last name pair.

@pdpinch @giocalitri

### CODE REVIEW

**Comment by rhysyngsun** [comment]
@amir-qayyum-khan realized I took this but it doesn't have a `Needs Review` tag. Let me know if that's the case and I'll give it a review!

**Comment by pdpinch** [comment]
I'd like to get @roberthouse54 to weigh in on the preferred name question before we merge this.

**Comment by roberthouse54** [comment]
I am ok with showing the name in the format FirstName LastName (preferredName).

**Comment by amir-qayyum-khan** [comment]
@aliceriot @pdpinch @roberthouse54 

screen shot 2016-11-22 at 2 42 08 pm

**Comment by gsidebo** [comment]
@roberthouse54 @pdpinch design question: should we hide this preferred name if it's the exact same as the user's first name? it seems unnecessary for us to show, for example, "Gavin Sidebottom (Gavin)"

**Comment by amir-qayyum-khan** [comment]
this  means that his name is Gavin Sidebottom (Gavin) but he prefer Gavin only.
i think this is ok.

File: static/js/util/util.js [hunk 1]
```diff
@@ -428,3 +428,14 @@ export function findCourseRun(
...
+
+/**
+ * Return first and last names if available else username
+ */
+export function getUserDisplayName(profile: Profile): string {
+  let first = profile.first_name || profile.username;
+  let last =  profile.last_name || '';
+  let preferred_name = profile.preferred_name ? `(${profile.preferred_name})` : '';
+
+  return `${first} ${last} ${preferred_name}`;
```
Comment (gsidebo) [inline static/js/util/util.js]: nitpick: can you make this `${first} ${last}${preferred_name}` and move the space into the `preferred_name` value so we don't have an extra space?

File: static/js/util/util_test.js [hunk 2]
```diff
@@ -538,4 +539,30 @@ describe('utility functions', () => {
...
+    beforeEach(() => {
+      profile = {
+        username: 'jane_username',
+        first_name: 'jane',
+        last_name: 'doe',
+        preferred_name: 'test'
+      };
+    });
+
+    it('assert first and last name', () => {
```
Comment (gsidebo) [inline static/js/util/util_test.js]: please change this `it` text to: `shows first, last, and preferred names`

File: static/js/util/util_test.js [hunk 3]
```diff
@@ -538,4 +539,30 @@ describe('utility functions', () => {
...
+      assert.equal('jane doe (test)', getUserDisplayName(profile));
+    });
+
+    it('returns username when first not available', () => {
```
Comment (gsidebo) [inline static/js/util/util_test.js]: please change this `it` text to: `shows username when first name is blank`

File: static/js/util/util_test.js [hunk 4]
```diff
@@ -538,4 +539,30 @@ describe('utility functions', () => {
...
+      profile.first_name = null;
+      assert.equal('jane_username doe (test)', getUserDisplayName(profile));
+    });
+
+    it('not return preferred_name when preferred_name not available', () => {
```
Comment (gsidebo) [inline static/js/util/util_test.js]: please change this `it` text to: `does not show preferred_name when that value is blank`

**Comment by amir-qayyum-khan** [comment]
Done @gsidebo

**Comment by roberthouse54** [comment]
We have spent probably too much time on this in my opinion, and I think part of the problem is that this is a required field, when it probably shouldn't be. The point of this field is really to allow users to specify a Preferred name if it is different from their first name. Probably in the majority of cases, the first name is going to be the preferred name, but we added this field in the cases where it is not. I do agree that this looks odd. But I think we are going to have problems like this as long as fields which should be optional are required. I think I agree with Gavin's suggestion as a solution for the time being.

**Comment by amir-qayyum-khan** [comment]
Done @roberthouse54 @gsidebo

**Pull request merged**